reference/forms/types/choice.rst

```diff
@@ -83,6 +92,28 @@ code, while the **key** is what will be shown to the user.
     and will be removed in 3.0. To read about the old API, read an older version of
     the docs.
 
+.. note::
+
+    Pre selected choices will depend on the **data** passed to field and the values of
+    the ``choices`` option. However submitted choices will depend on the **string**
+    matching the **choice**. In the example above, the default values falls back on a numeric
+    incrementation because ``null`` cannot be casted to string.
+    You should consider it as well when dealing with ``emtpy_data`` option::
+
+        $builder->add('isAttending', 'choice', array(
+            'choices'  => array(
+                'Maybe' => null,
+                'Yes' => true,
+                'No' => false,
+            ),
+            'choices_as_values' => true,
+            'data' => true, // pre selected choice
+            'empty_data' => '1', // default submitted value
+        ));
```

Comment:
Man, this note basically doesn't make any sense to me :). I don't think it's your fault - I just find this choice stuff really hard. What exactly is the problem that we're trying to warn the user about? The only issue I can understand is that `empty_data` (if you want to set that) would need to equal `1` to simulate `true` (I think, because apparently these values will render as 0, 1, 2 because they can't be cast into strings?).

I'm normally under the impression that the user will rarely care about the HTML value attribute that is rendered/submitted - as long as if I select "Yes", I end up with `true` in my code.